dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/Application/Authentication/AuthenticationOptions.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Microsoft.Teams.AI.Application.Authentication
+{
+    /// <summary>
+    /// Options for authentication.
+    /// </summary>
+    public class AuthenticationOptions
+    {
+        /// <summary>
+        /// The authentication classes to sign-in and sign-out users.
+        /// Key uniquely identifies each authentication.
+        /// </summary>
+        public Dictionary<string, IAuthentication> Authentications { get; set; }
+
+        /// <summary>
+        /// Describes the authentication class the bot should use if the user does not specify a authentication class name.
+        /// </summary>
+        public string? Default { get; set; }
+
+        /// <summary>
+        /// Indicates whether the bot should start the sign in flow when the user sends a message to the bot or triggers a message extension.
+        /// If the selector returns false, the bot will not start the sign in flow before routing the activity to the bot logic.
+        /// </summary>
+        public RouteSelector? AutoSignIn { get; set; }
```

Comment:
in JS this is just "Selector" which is the same type as "RouteSelector" but just a different name to avoid confusion.

Reply:
Thanks. Defined new `SelectorAsync` delegate for the option.